docs/articles/rag-application-communication-system.md

```diff
@@ -0,0 +1,164 @@
+
+Dissociation between use and design should be a primary concern. Without proper care, LLMs and embedding models are not only unable to solve existing data frictions but will introduce new frictions : alignment mismatch with user expectation, poor breakdown of data structure due to tokenization, incongruous length, total lack of control over the system purpose (the infamous “Chevrolet” effect).
+
+### 2.3 Know your data
+Like your users, the data will not necessarily agree with the constraints of language models and embedding models.
+
+Context window is an obvious restriction. Despite its gradual lengthening, the indexation of long texts always underperforms. This is one of the biggest mismatches between pre-training data and data in production: the vast majority of Common Crawl texts are less than 512 tokens. Organizations routinely deal with multi-pages PDF.
+
+So you need to split or chunk your document and… this is not trivial. Documents parts are not meant to be standalone units of content and disambiguation requires some level of cross-referencing. The biggest performance gain on my retrieval evaluation was the… systematic inclusion of title hierarchy (document > chapter > section etc.). Simply because in long documents, general information on the topic is given at this scale, but not recalled in specific paragraphs.
+
+Proper contextual support for original data is not only a concern for retrieval but also for generation. Even powerful LLMs are heavily sensitive to document structure. Lack of proper segmentation or unusual structure will result in poor tokenization (typically merging a document separator with a token) and a higher rate of omissions and hallucinations. With tables, a hacky (but verbose) fix I’ve found is to recall the row name/column name couple before each entry.
+
+Finally, retrieval metrics are not necessarily fitting the data. Embedding models rely on a fuzzy concept of “similarity” that is likely to fail the last mile. They are trained on specific sets of documents and questions (typically bases like MS-Marco). This will not necessarily extend well to specialized resources, and, more fundamentally, similarity may not align with user expectations.
+
+### 2.4 Settling for hybrid
+As RAG systems naturally evolve with longer documents, enriched with additional context and more and more uninitiated users, simply transposing their usual search habits, the reality sets in: embedding models are barely competitive to classic information retrieval approaches like BM25 and have to be supplemented by it.
+
```

Comment:
1. "embedding models" in RAG systems?
2. "supplemented by it" = embedding models have to be supplemented by classic info retrieval approaches?

Reply:
Yes on both count.